### PR TITLE
fix(GraphQL): change Default client to Client with timeout for fetching JWKs

### PR DESCRIPTION
**Pull request by minhaj-shakeel**
Fixes GRAPHQL-790.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6891)

### CODE REVIEW

File: graphql/schema/schemagen.go [hunk 1]
```diff
@@ -221,8 +221,9 @@ func NewHandler(input string, validateOnly bool) (Handler, error) {
 	}
 
 	// If Dgraph.Authorization header is parsed successfully and JWKUrl is present
-	// then Fetch the JWKs from the JWKUrl
+	// then initialise the http client and Fetch the JWKs from the JWKUrl
 	if metaInfo != nil && metaInfo.JWKUrl != "" {
+		metaInfo.NewClient()
```
Comment (aman-bansal) [inline graphql/schema/schemagen.go:226]: reuse the connection. No need to create the client again and again

File: graphql/authorization/auth.go [hunk 2]
```diff
@@ -439,7 +441,7 @@ func (a *AuthMeta) FetchJWKs() error {
 		return err
 	}
 
-	resp, err := http.DefaultClient.Do(req)
+	resp, err := a.httpClient.Do(req)
```
Comment (aman-bansal) [inline graphql/authorization/auth.go:444]: use `defer resp.Body.Close()`. In this way same connection would be available for reuse.

**Review by minhaj-shakeel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6891)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @aman-bansal, @MichaelJCompton, and @pawanrawal)

___
*[graphql/authorization/auth.go, line 444 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6891#-MM-CyxM7CI8U5HSFOmF-r1-444:-MM-r2ZR8BCUos9Khmnx:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/2deb2a5745b7aab580ae752563c103a97741f42e/graphql/authorization/auth.go#L444)):*
<details><summary><i>Previously, aman-bansal (aman bansal) wrote…</i></summary><blockquote>

use `defer resp.Body.Close()`. In this way same connection would be available for reuse.
</blockquote></details>

Done.
___
*[graphql/schema/schemagen.go, line 226 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6891#-MM-CyxM7CI8U5HSFOmG-r1-226:-MM-r3Bb3jPddjQ2GbT3:b-too0ka) ([raw file](https://github.com/dgraph-io/dgraph/blob/2deb2a5745b7aab580ae752563c103a97741f42e/graphql/schema/schemagen.go#L226)):*
<details><summary><i>Previously, aman-bansal (aman bansal) wrote…</i></summary><blockquote>

reuse the connection. No need to create the client again and again
</blockquote></details>

`NewHandler` is called only once when the new schema is pushed.

**Review by pawanrawal — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6891)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @aman-bansal, @MichaelJCompton, @minhaj-shakeel, and @pawanrawal)

___
*[graphql/authorization/auth.go, line 444 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6891#-MM-CyxM7CI8U5HSFOmF-r1-444:-MM0IShb0bGxBJaWDsEq:b-jg6ewz) ([raw file](https://github.com/dgraph-io/dgraph/blob/2deb2a5745b7aab580ae752563c103a97741f42e/graphql/authorization/auth.go#L444)):*
<details><summary><i>Previously, minhaj-shakeel wrote…</i></summary><blockquote>

Done.
</blockquote></details>

nice catch
___
*[graphql/authorization/auth.go, line 545 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/6891#-MM0J3NC4sK_lWIscyw_:-MM0J3NC4sK_lWIscywa:bujzn4g) ([raw file](https://github.com/dgraph-io/dgraph/blob/ffca389848aa60ddeadb6528b65528deab474256/graphql/authorization/auth.go#L545)):*
> ```Go
> }
> 
> func (a *AuthMeta) NewClient() {
> ```

This can be called `InitHttpClient`

**Pull request merged**